File: profiles/chlcc/hud/commonmenu.lua
root.CommonMenu = {
  Type = CommonMenuType.CHLCC,
  CircleStartPosition = { X = 20, Y = 20 },
  CircleOffset = 200,

  ErinSprite = "Erin",
  ErinPosition = { X = 301, Y = 1 },
  BackgroundFilter = "BackgroundFilter",

  InitialRedBarPosition = { X = 0, Y = 538 },
  RightRedBarPosition = { X = 1059, Y = 538 },
  RedBarDivision = 1826,
  RedBarBaseX = 1059,
  RedBarSprite = "RedBar",
  RedBarLabelPosition = { X = 1067, Y = 573 },
  RedBarLabel = "RedBarLabel",

  TitleFadeInDuration = 40 / 60,
  TitleFadeOutDuration = 28 / 60,
  MenuSelectPromptDuration =  110 / 60,
  MenuSelectPromptInterval = 5 / 60,
  MenuTransitionDuration = 64 / 60,
  ShowPageAnimationStartTime = 16 / 60,
  ShowPageAnimationDuration = (48 - 16) / 60,
  ButtonPromptAnimationStartTime = 48 / 60,
  ButtonPromptAnimationDuration = (64 - 48) / 60,
  ButtonPromptStartPosition = { X = 603 + (64 - 48) * 40, Y = 651 },
  DiagonalTitlesOffsetStart = {X = 572.0 * 3, Y= -460.0 },
  DiagonalTitlesOffsetEnd = {X = -572.0 , Y= 460.0 / 3.0 }
};

root.Sprites["Erin"] = {
    Sheet = "Main",
    Bounds = { X = 641, Y = 1, Width = 978, Height = 798 }
}

root.Sprites["BackgroundFilter"] = {
    Sheet = "Main",
    Bounds = { X = 0, Y = 0, Width = 640, Height = 360 }
}

root.Sprites["RedBar"] = {
    Sheet = "Main",
    Bounds = { X = 767, Y = 913, Width = 1280, Height = 110 }
}

root.Sprites["RedBarLabel"] = {
    Sheet = "Main",
    Bounds = { X = 506, Y = 469, Width = 133, Height = 74 }
}

File: src/profile/games/chlcc/commonmenu.cpp
#include "commonmenu.h"
#include "../../profile_internal.h"

namespace Impacto {
namespace Profile {
namespace CHLCC {
namespace CommonMenu {

void Configure() {
  TitleFadeInDuration = EnsureGetMember<float>("TitleFadeInDuration");
  TitleFadeOutDuration = EnsureGetMember<float>("TitleFadeOutDuration");

  CircleStartPosition = EnsureGetMember<glm::vec2>("CircleStartPosition");
  CircleOffset = EnsureGetMember<float>("CircleOffset");

  ErinPosition = EnsureGetMember<glm::vec2>("ErinPosition");
  ErinSprite = EnsureGetMember<Sprite>("ErinSprite");

  BackgroundFilter = EnsureGetMember<Sprite>("BackgroundFilter");

  InitialRedBarPosition = EnsureGetMember<glm::vec2>("InitialRedBarPosition");
  RightRedBarPosition = EnsureGetMember<glm::vec2>("RightRedBarPosition");
  InitialRedBarSprite = EnsureGetMember<Sprite>("RedBarSprite");
  RedBarDivision = EnsureGetMember<float>("RedBarDivision");
  RedBarBaseX = EnsureGetMember<float>("RedBarBaseX");
  RedBarLabelPosition = EnsureGetMember<glm::vec2>("RedBarLabelPosition");
  RedBarLabel = EnsureGetMember<Sprite>("RedBarLabel");

  MenuSelectPromptDuration = EnsureGetMember<float>("MenuSelectPromptDuration");
  MenuSelectPromptInterval = EnsureGetMember<float>("MenuSelectPromptInterval");
  MenuTransitionDuration = EnsureGetMember<float>("MenuTransitionDuration");
  ShowPageAnimationStartTime =
      EnsureGetMember<float>("ShowPageAnimationStartTime");
  ShowPageAnimationDuration =
      EnsureGetMember<float>("ShowPageAnimationDuration");
  ButtonPromptAnimationStartTime =
      EnsureGetMember<float>("ButtonPromptAnimationStartTime");
  ButtonPromptAnimationDuration =
      EnsureGetMember<float>("ButtonPromptAnimationDuration");
  ButtonPromptStartPosition =
      EnsureGetMember<glm::vec2>("ButtonPromptStartPosition");
  DiagonalTitlesOffsetStart =
      EnsureGetMember<glm::vec2>("DiagonalTitlesOffsetStart");
  DiagonalTitlesOffsetEnd =
      EnsureGetMember<glm::vec2>("DiagonalTitlesOffsetEnd");
}

}  // namespace CommonMenu
}  // namespace CHLCC
}  // namespace Profile
}  // namespace Impacto